diff --git a/ref/Makefile b/ref/Makefile
index 7004b734..917505dc 100644
--- a/ref/Makefile
+++ b/ref/Makefile
@@ -61,14 +61,14 @@ libpqcrystals_aes256ctr_ref.so: aes256ctr.c aes256ctr.h
 libpqcrystals_sha2_ref.so: sha256.c sha512.c sha2.h
 	$(CC) -shared -fPIC $(CFLAGS) sha256.c sha512.c -o libpqcrystals_sha2_ref.so
 
-libpqcrystals_kyber512_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 $(SOURCES) symmetric-shake.c -o libpqcrystals_kyber512_ref.so
+libpqcrystals_kyber512_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 $(SOURCES) fips202.c randombytes.c symmetric-shake.c -o libpqcrystals_kyber512_ref.so
 
-libpqcrystals_kyber768_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 $(SOURCES) symmetric-shake.c -o libpqcrystals_kyber768_ref.so
+libpqcrystals_kyber768_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 $(SOURCES) randombytes.c fips202.c symmetric-shake.c -o libpqcrystals_kyber768_ref.so
 
-libpqcrystals_kyber1024_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 $(SOURCES) symmetric-shake.c -o libpqcrystals_kyber1024_ref.so
+libpqcrystals_kyber1024_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c aes256ctr.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 $(SOURCES) aes256ctr.c randombytes.c fips202.c symmetric-shake.c -o libpqcrystals_kyber1024_ref.so
 
 test_kyber512: $(SOURCESKECCAK) $(HEADERSKECCAK) test_kyber.c randombytes.c
 	$(CC) $(CFLAGS) -DKYBER_K=2 $(SOURCESKECCAK) randombytes.c test_kyber.c -o test_kyber512
@@ -106,14 +106,14 @@ test_speed768: $(SOURCESKECCAK) $(HEADERSKECCAK) cpucycles.h cpucycles.c speed_p
 test_speed1024: $(SOURCESKECCAK) $(HEADERSKECCAK) cpucycles.h cpucycles.c speed_print.h speed_print.c test_speed.c randombytes.c
 	$(CC) $(CFLAGS) -DKYBER_K=4 $(SOURCESKECCAK) randombytes.c cpucycles.c speed_print.c test_speed.c -o test_speed1024
 
-libpqcrystals_kyber512-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 -DKYBER_90S $(SOURCES) symmetric-aes.c -o libpqcrystals_kyber512-90s_ref.so
+libpqcrystals_kyber512-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c aes256ctr.c randombytes.c sha2.h sha256.c sha512.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 -DKYBER_90S $(SOURCES) sha512.c sha256.c randombytes.c aes256ctr.c symmetric-aes.c -o libpqcrystals_kyber512-90s_ref.so
 
-libpqcrystals_kyber768-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 -DKYBER_90S $(SOURCES) symmetric-aes.c -o libpqcrystals_kyber768-90s_ref.so
+libpqcrystals_kyber768-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c aes256ctr.c randombytes.c sha2.h sha256.c sha512.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 -DKYBER_90S $(SOURCES) sha512.c sha256.c randombytes.c aes256ctr.c symmetric-aes.c -o libpqcrystals_kyber768-90s_ref.so
 
-libpqcrystals_kyber1024-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 -DKYBER_90S $(SOURCES) symmetric-aes.c -o libpqcrystals_kyber1024-90s_ref.so
+libpqcrystals_kyber1024-90s_ref.so: $(SOURCES) $(HEADERS) symmetric-aes.c aes256ctr.c randombytes.c sha2.h sha256.c sha512.c
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 -DKYBER_90S $(SOURCES) sha512.c sha256.c randombytes.c aes256ctr.c symmetric-aes.c -o libpqcrystals_kyber1024-90s_ref.so
 
 test_kyber512-90s: $(SOURCESNINETIES) $(HEADERSNINETIES) test_kyber.c randombytes.c
 	$(CC) $(CFLAGS) -D KYBER_90S -DKYBER_K=2 $(SOURCESNINETIES) randombytes.c test_kyber.c -o test_kyber512-90s

