diff --git a/ref/Makefile b/ref/Makefile
index c0bcbd4..5c9ce81 100644
--- a/ref/Makefile
+++ b/ref/Makefile
@@ -42,17 +42,17 @@ lib/libpqcrystals_fips202_ref.so: fips202.c fips202.h
 	mkdir -p lib
 	$(CC) -shared -fPIC $(CFLAGS) fips202.c -o $@
 
-lib/libpqcrystals_kyber512_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
+lib/libpqcrystals_kyber512_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c
 	mkdir -p lib
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 $(SOURCES) symmetric-shake.c -o $@
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=2 $(SOURCES) fips202.c randombytes.c symmetric-shake.c -o $@
 
-lib/libpqcrystals_kyber768_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
+lib/libpqcrystals_kyber768_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c
 	mkdir -p lib
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 $(SOURCES) symmetric-shake.c -o $@
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=3 $(SOURCES) fips202.c randombytes.c symmetric-shake.c -o $@
 
-lib/libpqcrystals_kyber1024_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c
+lib/libpqcrystals_kyber1024_ref.so: $(SOURCES) $(HEADERS) symmetric-shake.c fips202.c randombytes.c
 	mkdir -p lib
-	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 $(SOURCES) symmetric-shake.c -o $@
+	$(CC) -shared -fPIC $(CFLAGS) -DKYBER_K=4 $(SOURCES) fips202.c randombytes.c symmetric-shake.c -o $@
 
 test/test_kyber512: $(SOURCESKECCAK) $(HEADERSKECCAK) test/test_kyber.c randombytes.c
 	$(CC) $(CFLAGS) -DKYBER_K=2 $(SOURCESKECCAK) randombytes.c test/test_kyber.c -o $@
